### PR TITLE
[INLONG-7619][Manager] Support update and retry MySQL sources after updating MySQLDataNode

### LINKED ISSUE #7619: [Improve][Manager] Support retry mysqlSources after updating MysqlDataNode
### Description

_No response_

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #7619 

### Motivation

Support retry mysqlSources after updating MysqlDataNode.

### Modifications

Support retry mysqlSources after updating MysqlDataNode.

### CODE REVIEW

**Review by vernedeng — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-dao/src/main/resources/mappers/StreamSourceEntityMapper.xml [hunk 1]
```diff
@@ -319,7 +319,24 @@
...
         </where>
     </select>
-
+    <select id="selectByClusterAndDataNode" resultType="org.apache.inlong.manager.dao.entity.StreamSourceEntity">
+        select
+        <include refid="Base_Column_List"/>
+        from stream_source
+        <where>
+            is_deleted = 0
+            and status not in (99, 110)
```
Comment (healchow) [inline inlong-manager/manager-dao/src/main/resources/mappers/StreamSourceEntityMapper.xml:328]: Why filter those status?
Can we description it on the method name, such as `selectNormalXxx` or `selectUnDeletedXxx`.

File: inlong-manager/manager-dao/src/main/resources/mappers/StreamSourceEntityMapper.xml [hunk 2]
```diff
@@ -458,6 +475,22 @@
...
+        update stream_source
+        <set>
+            previous_status = status,
+            status          = #{status, jdbcType=INTEGER},
+            modifier        = #{operator,jdbcType=VARCHAR},
+            version         = version + 1
+        </set>
+        <where>
+            is_deleted = 0
+            and id in
```
Comment (healchow) [inline inlong-manager/manager-dao/src/main/resources/mappers/StreamSourceEntityMapper.xml:488]: How about the `idList` was empty?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/AbstractDataNodeOperator.java [hunk 3]
```diff
@@ -95,7 +110,41 @@ public Boolean testConnection(DataNodeRequest request) {
...
     }
+
+    public void retryStreamSourceByDataNodeNameAndType(String dataNodeName, String type, String operator) {
+        Integer status = SourceStatus.TO_BE_ISSUED_RETRY.getCode();
+        LOGGER.info("begin to update stream source status by dataNodeName={}, status={}, by operator={}",
+                dataNodeName, status, operator);
+        List<StreamSourceEntity> streamSourceEntities = sourceMapper.selectByClusterAndDataNode(null,
+                dataNodeName, type);
+        List<Integer> needUpdateIds = new ArrayList<>();
+        for (StreamSourceEntity source : streamSourceEntities) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/AbstractDataNodeOperator.java:124]: The efficiency of this for loop is very low, can the filter be completed in the above query?

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/StreamSourceEntityMapper.java [hunk 4]
```diff
@@ -148,6 +154,14 @@ int updateIpAndUuid(@Param("id") Integer id, @Param("agentIp") String agentIp, @
 
     int updateSnapshot(StreamSourceEntity entity);
 
+    /**
+     * @param idList db source id list
```
Comment (healchow) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/StreamSourceEntityMapper.java:158]: There are comments for `@param` tag, but no Javadoc fot this method.

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/StreamSourceEntityMapper.java [hunk 5]
```diff
@@ -148,6 +154,14 @@ int updateIpAndUuid(@Param("id") Integer id, @Param("agentIp") String agentIp, @
...
+     * @param status modify status
+     * @param operator operator name
```
Comment (healchow) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/StreamSourceEntityMapper.java:160]:
```suggestion
     * Update the source status
     *
     * @param idList source id list
     * @param status modify the status to this
     * @param operator operator name
```

File: inlong-manager/manager-dao/src/main/resources/mappers/StreamSourceEntityMapper.xml (hunk 1 above)
Comment (fuweng11) [reply inlong-manager/manager-dao/src/main/resources/mappers/StreamSourceEntityMapper.xml]: done

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/StreamSourceEntityMapper.java (hunk 5 above)
Comment (fuweng11) [reply inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/mapper/StreamSourceEntityMapper.java]: done

**Review by baomingyu — APPROVED** [review]
LGTM

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**